### PR TITLE
sof-test: split both type pipeline into playback and capture

### PR DESCRIPTION
**Pull request by aiChaoSONG**
1. split both type pipeline into playback and capture pipelines
2.  remove "both" in test cases

@Bin-QA Wubin will modify the simultaneous-playback-capture.sh test case, and add logic to extract "both" type pipeline.

***DO NOT MERGE BEFORE PR #10 ***

### CODE REVIEW

File: tools/sof-tplgreader.py [hunk 1]
```diff
@@ -46,6 +59,25 @@ def loadFile(self, filename, sofcard=0):
                 pipeline_dict['fmt'] = pipeline_dict['fmts'].split(' ')[0]
                 pipeline_dict['rate_min'], pipeline_dict['rate_max'] = self._key2str(cap, 'rate')
                 pipeline_dict['ch_min'], pipeline_dict['ch_max'] = self._key2str(cap, 'channels')
+                if pcm_type == "both":
+                    pipeline_dict["type"] = "capture"
+                    # copy pipeline and change values
+                    pb_pipeline_dict = copy.copy(pipeline_dict)
+                    pb_pipeline_dict["type"] = "playback"
+                    cap = pcm["caps"][pcm['playback']]
+                    pga = pgas[pcm['capture']]
```
Comment (Bin-QA) [inline tools/sof-tplgreader.py]: Different value between 'playback' and 'capture' ?

File: tools/sof-tplgreader.py [hunk 2]
```diff
@@ -46,6 +59,25 @@ def loadFile(self, filename, sofcard=0):
                 pipeline_dict['fmt'] = pipeline_dict['fmts'].split(' ')[0]
                 pipeline_dict['rate_min'], pipeline_dict['rate_max'] = self._key2str(cap, 'rate')
                 pipeline_dict['ch_min'], pipeline_dict['ch_max'] = self._key2str(cap, 'channels')
+                if pcm_type == "both":
+                    pipeline_dict["type"] = "capture"
+                    # copy pipeline and change values
+                    pb_pipeline_dict = copy.copy(pipeline_dict)
```
Comment (Bin-QA) [inline tools/sof-tplgreader.py]: I saw you rebuild the 'pga' and 'eq' don't need deep copy, you can direct to use `pipeline_dict.copy`

File: tools/sof-tplgreader.py [hunk 3]
```diff
@@ -46,6 +59,25 @@ def loadFile(self, filename, sofcard=0):
...
+                        pga_names = [i['name'] for i in pga]
+                        pipeline_dict['pga'] = " ".join(pga_names)
+                    eq = eqs[pcm['capture']]
+                    if eq != []:
+                        eq_names = [i['name'] for i in eq]
+                        pipeline_dict['eq'] = " ".join(eq_names)
+                    pb_pipeline_dict["fmts"] = " ".join(formats[pcm['playback']])
+                    pipeline_dict['fmt'] = pipeline_dict['fmts'].split(' ')[0]
+                    pb_pipeline_dict['rate_min'], pb_pipeline_dict['rate_max'] = self._key2str(cap, 'rate')
+                    pb_pipeline_dict['ch_min'], pb_pipeline_dict['ch_max'] = self._key2str(cap, 'channels')
```
[the remark this answers is not in the document]
Comment (aiChaoSONG) [reply tools/sof-tplgreader.py]: @Bin-QA  in fact, these values for playback and capture are from their respectively caps structure under pcm structure. values under pcm structure is the common part, but for values under caps structure, we should use their own values, through they may be the same value currently. I mean, they may not be the same in the future.

File: tools/sof-tplgreader.py (hunk 1 above)
Comment (aiChaoSONG) [reply tools/sof-tplgreader.py]: yes, PGA and EQ is total the different value for playback and capture

File: tools/sof-tplgreader.py (hunk 2 above)
Comment (aiChaoSONG) [reply tools/sof-tplgreader.py]: sure, will refine in next push

**Pull request merged**